server/db/driver/pg/internal/matches.go

```diff
@@ -114,6 +114,82 @@ const (
 	WHERE (takerAccount = $1 OR makerAccount = $1)
 		AND active;`
 
+	// TallyInactiveUserMatches gets a tally of inactive matches for a user,
+	// grouped by match status. This excludes cancel order matches with
+	// "takerSell IS NOT NULL". Since matches in MatchComplete status may still
+	// be labeled active if awaiting aSigAckOfBRedeem, all matches in
+	// MatchComplete are returned regardless of the active flag. MatchComplete
+	// is inferred by "bRedeemTime IS NOT NULL" so the MatchComplete integer
+	// value need not be a query argument.
+	TallyInactiveUserMatches = `SELECT COUNT(*), status, makeraccount=$1 AS isMaker, takeraccount=$1 AS isTaker
+	FROM %s
+	WHERE takerSell IS NOT NULL
+		AND (takerAccount = $1 OR makerAccount = $1)
+		AND (NOT active OR bRedeemTime IS NOT NULL)
+	GROUP BY status, isMaker, isTaker;`  // separate rows for: maker, taker, both
+
+	// TallyInactiveUserMatchesLastN breaks down the user's inactive matches.
+	// This is likely to be useful for administrative review of users.
+	// e.g.
+	// count | status | isMaker | isTaker
+	// -------+--------+---------+---------
+	//    10 |      0 | t       | f             <-- at fault (maker in NewlyMatched)
+	//     1 |      1 | f       | t             <-- at fault (taker in MakerSwapCase)
+	//     2 |      1 | t       | f             <-- not at fault (maker in MakerSwapCast)
+	//     8 |      2 | t       | f             <-- at fault (maker in TakerSwapCast)
+	//     2 |      2 | t       | t             <-- at fault (also maker in TakerSwapCast, but taker too)
+	//     3 |      3 | t       | f             <-- not at fault (maker in MakerRedeemed)
+	//     7 |      4 | f       | t             <-- success as taker
+	//   125 |      4 | t       | f             <-- success as maker
+	//   101 |      4 | t       | t             <-- success as both
+	TallyInactiveUserMatchesLastN = `SELECT COUNT(*), status, isMaker, isTaker
+	FROM (
+		SELECT status, makerAccount = $1 AS isMaker, takerAccount = $1 AS isTaker
+		FROM %s
+		WHERE takerSell IS NOT NULL
+			AND (NOT active OR bRedeemTime IS NOT NULL)
+			AND (makerAccount = $1 OR takerAccount = $1)
+		ORDER BY GREATEST(epochIdx*epochDur, aContractTime, bContractTime, aRedeemTime, bRedeemTime) DESC  -- last action time i.e. success or approx. when could have acted
+		LIMIT $2
+	) AS matches
+	GROUP BY status, isMaker, isTaker;`
+
+	InactiveUserMatchesLastN = `
+		SELECT status, makerAccount = $1 AS isMaker, takerAccount = $1 AS isTaker,
+			GREATEST(epochIdx*epochDur, aContractTime, bContractTime, aRedeemTime, bRedeemTime) AS lastTime
+		FROM %s
+		WHERE takerSell IS NOT NULL
+			AND (NOT active OR bRedeemTime IS NOT NULL)
+			AND (makerAccount = $1 OR takerAccount = $1)
+		ORDER BY lastTime DESC  -- last action time i.e. success or approx. when could have acted
+		LIMIT $2;`
+
+	CompletedOrAtFaultMatchesLastN = `
+		WITH acct (aid) AS ( VALUES($1::BYTEA) )
+
+		SELECT status, (status=4 OR (status=3 AND makerAccount = aid)) AS success,
```

Comment:
I was not happy about the compromise either.  I thought I made a comment somewhere to note this, but I will add one now.

Reply:
Could used a named string constant and string concatenation. Then check that value is the same as the expected order.OrderStatus int value in `init()` and panic if not. That would allow it to stay a constant and also not compile if different than whats in "order".

Reply:
I like that idea, thanks.  Will likely go that route.